### PR TITLE
chore: noirjs integration testing

### PR DESCRIPTION
**Pull request by jonybur**
# Description

- Updates integration testing to use noirjs instead of individual packages
- Update ACVM tracking workflow

## Problem\*

Resolves https://github.com/noir-lang/noir/issues/2738

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by jonybur** [comment]
Successful execution: https://github.com/noir-lang/noir/actions/runs/6224837507/job/16893921657

File: .github/workflows/track-acvm.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
         run: |
           # Extract acvm version from the current Cargo.lock
           CURRENT_VERSION=$(awk '/name = "acvm"/ {getline; print $3}' Cargo.lock | tr -d '"')
-          INTEGRATION_TEST_VERSION=$(yarn workspace integration-tests info @noir-lang/acvm_js --json | jq .children.Version | tr -d '"')
+          INTEGRATION_TEST_VERSION=$(awk '/@noir-lang\/noir_js@workspace:/ {flag=1; next} /noirc_abi: "workspace:*/ {flag=0} flag' yarn.lock | grep -Eo '"@noir-lang/acvm_js": [0-9.]+[^,]' | awk -F': ' '{print $2}' | tr -d ' "')
```
Comment (TomAFrench) [inline .github/workflows/track-acvm.yml:20]: What's the benefit of this?

Comment (jonybur) [reply .github/workflows/track-acvm.yml:20]: The integration test no longer has a version of acvm_js to pick up the version number from. I now get it from yarn.lock. I could however get it from noir_js package.json. Thoughts?

File: compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts [hunk 2]
```diff
@@ -4,18 +4,17 @@ import newCompiler, {
...
-} from "@noir-lang/acvm_js";
-
 import { Barretenberg, RawBuffer, Crs } from "@aztec/bb.js";
+import { acvm, noirc } from "@noir-lang/noir_js";
+import { decompressSync as gunzip } from "fflate";
 
 import * as TOML from "smol-toml";
 
+const { default: initACVM, executeCircuit, compressWitness } = acvm;
+const { default: newABICoder, abiEncode } = noirc;
```
Comment (TomAFrench) [inline compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts:14]: Not relevant for this PR but we should look into setting up package exports so we can expose these as

```
import newABICoder, { abiEncode } from "@noir-lang/noir_js/abi";
import initACVM, {
  executeCircuit,
  WitnessMap,
  compressWitness,
} from "@noir-lang/noir_js/acvm";
```

File: .github/workflows/track-acvm.yml (hunk 1 above)
Comment (TomAFrench) [reply .github/workflows/track-acvm.yml:20]: Yeah, pulling it from `noir_js` would be my preference on this. All this `awk`ing isn't particularly readable.

Comment (jonybur) [reply .github/workflows/track-acvm.yml]: Please recheck

**Review by TomAFrench — APPROVED** [review]
LGTM

**Pull request merged**